src/PSR7/Validator.php

```diff
@@ -41,6 +51,15 @@ public static function fromYaml(string $yaml, ?CacheItemPoolInterface $cache = n
         return new static($oas);
     }
 
+    /**
+     * @param string $json
+     * @param CacheItemPoolInterface|null $cache
+     *
+     * @return static
+     *
+     * @throws InvalidArgumentException
```

Comment:
This is autogenerated since thrown by `cachedRead --  cache->getItem` call.

Reply:
Yeah, there are different types of logic encapsulated, different exceptions to expect.